NumberTheory/FastPower.cpp

```diff
@@ -16,23 +16,24 @@
 
 */
 #include <iostream>
+#include <cmath>
 
 using namespace std;
 typedef unsigned long long ull;
 
 //Function that returns base raised to the exponent
-ull fast_power (ull base,ull exponent)
+ull fast_power (ull base,ull exponent,ull mod)
```

Comment:
Give `mod` the default value `ULLONG_MAX` so that it doesn't need to be passed to the function if modulus of the result is not required.